JS/Lesson16/savas/script.js

```diff
@@ -1,75 +1,173 @@
 // 8. Write a while loop that generates random numbers between 1 and 10 until a number greater than 8 is generated, and then prints out the total number of iterations.
+function getRandomNumber () {
+	return Math.floor(Math.random()*10)+1;
+}
+
+let iteration = 1;
+let randomNumber = getRandomNumber();
+let randomNumbers = [randomNumber];
+while(randomNumber <= 8) {
+	randomNumber = getRandomNumber();
+	randomNumbers.push(randomNumber);
+	iteration++;
+}
+
+console.log("Iterations:", iteration, "Random Number:", randomNumber, "Numbers:", randomNumbers);
 
-console.log('- - - - - - - - - - - - - - - - - ');
+console.log('- - - - - - - - - - - - - - - - - EXERCISE 9 - - - - - - - - - - - - - - - - - ');
 // 9. Write a do-while loop that counts from 1 to 10 and prints out each number.
+let number = 1;
+do {
+	console.log(number);
+	number++;
+}
+
+while (number <= 10);
 
-console.log('- - - - - - - - - - - - - - - - - ');
+
+console.log('- - - - - - - - - - - - - - - - - EXERCISE 10 - - - - - - - - - - - - - - - - ');
 // 10. Write a for...in loop that prints out the keys and values of an object.
 const person = {name: "John", surname: "Doe", age: 30, city: "New York"};
 // Array olsaydi boyle olurdu: ["John", "Doe", 30 , "New York"];
 // Array element position numbers: 0, 1, 2, 3
 // Object keys: name, surname, age, city 
 
-console.log('- - - - - - - - - - - - - - - - - ');
+for (const objectKey in person) {
+	console.log(`${objectKey}: ${person[objectKey]}`);
+}
+
+
+console.log('- - - - - - - - - - - - - - - - - EXERCISE 11 - - - - - - - - - - - - - - - - ');
 // 11. Write a for...in loop that sums the values of an object.
 const obj = {a: 10, b: 20, c: 30, d: 23};
+let sum = 0;
+for (const key in obj) {
+	sum += obj[key];
+}
+
+console.log(sum);
 
-console.log('- - - - - - - - - - - - - - - - - ');
+console.log('- - - - - - - - - - - - - - - - - EXERCISE 12 - - - - - - - - - - - - - - - - ');
 // 12. Write a loop that finds the average of an array of numbers.
 const numbers12 = [10, 20, 30, 40, 50];
 
-console.log('- - - - - - - - - - - - - - - - - ');
+let sum1 = 0;
+
+for (i = 0; i < numbers12.length; i++) {
+	sum1 += numbers12[i];
+}
+
+let average = sum / numbers12.length;
+
+console.log(`The average of the numbers is ${average}.`);
+
+console.log('- - - - - - - - - - - - - - - - - EXERCISE 13 - - - - - - - - - - - - - - - - ');
 // 13. Write a loop that prints out the sum of all even numbers between 1 and 50.
+let sum2 = 0;
 
+for (i = 0; i <= 50; i += 2) {
+	sum2 += i;
+}
 
+console.log(`The sum of all even numbers between 1 and 50 is ${sum2}.`);
 
-console.log('- - - - - - - - - - - - - - - - - ');
+console.log('- - - - - - - - - - - - - - - - - EXERCISE 14 - - - - - - - - - - - - - - - - ');
 // 14. Write a loop that finds the second largest number in an array of numbers.
-const numbers14 = [10, 5, 20, 8, 15];
-
-console.log('- - - - - - - - - - - - - - - - - ');
+let numbers14 = [10, 5, 20, 8, 15];
+numbers14.sort(function(a, b) {
```

Comment:
Smart solution! It's not really part of the "iterations" topic but it is something that's used more often in real word.

One thing that you need to be careful with is that there could be more than one of the same number in the array. E.g. for array like this: `[20, 5, 20, 8, 15];` the code doesn't work any more.